src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1958 @@
+        Tensor: xb_mask, masked input, same shape as input Tensor: Mask tensor of shape [bs x c x n] or [bs x tsg1 x
+        tsg2 x c x n]
+    """
+    if seed_number:
+        set_seed(seed_number)
+
+    if mix_ratio is None:
+        mix_ratio = [1 for t in patch_lengths]
+
+    batch_size, nvars, seq_len, feat = xb.shape
+    mask = torch.zeros(batch_size, nvars, seq_len, device=xb.device)
+
+    t_list = []
+    total_length = 0
+    total_ratio = sum(mix_ratio)
+
+    for i, j in zip(patch_lengths, mix_ratio):
+        if i <= 0 or i >= seq_len:
+            raise Exception("masked_patch_len should be greater than 0 and less than total patches.")
+        temp_len = int(batch_size * j / total_ratio)
+        t_list.append([i, j, temp_len])
+        total_length += temp_len
+
+    t_list = sorted(t_list, key=lambda x: x[2])
+
+    if total_length < batch_size:
+        t_list[0][2] = t_list[0][2] + (batch_size - total_length)
+    elif total_length > batch_size:
+        t_list[-1][2] = t_list[-1][2] + (total_length - batch_size)
+
+    b1 = 0
+    for p, _, l in t_list:
+        b2 = b1 + l
+        mask[b1:b2, :, -p:] = 1
+        b1 = b2
+
+    perm = torch.randperm(mask.shape[0])
+    mask = mask[perm]
+
+    mask = mask.unsqueeze(-1).repeat(1, 1, 1, feat)  # mask: [bs x nvars x num_patch x patch_len]
+    if unmasked_channel_indices is not None:
+        mask[:, unmasked_channel_indices, :, :] = 0
+
+    xb_mask = xb.masked_fill(mask.bool(), mask_value)
+    return xb_mask, mask[..., 0]
+
+
+def compute_num_patches(sequence_length, patch_length, stride):
+    return (max(sequence_length, patch_length) - patch_length) // stride + 1
+
+
+class Patchify(nn.Module):
+    """
+    A class to patchify the time series sequence into different patches
+
+    Parameters:
+        sequence_length (int, required): input sequence length.
+        patch_length (int, required): patch length.
+        stride (int, required): stride between patches.
+
+    Returns:
+        z: output tensor data [bs x num_input_channels x num_patches x patch_length]
+    """
+
+    def __init__(
+        self,
+        sequence_length: int,
+        patch_length: int,
+        stride: int,
+        padding: bool = False,  # TODO: use this to set whether we want to pad zeros to the sequence
+    ):
+        super().__init__()
+
+        assert (
+            sequence_length > patch_length
+        ), f"Sequence length ({sequence_length}) has to be greater than the patch length ({patch_length})"
+
+        self.sequence_length = sequence_length
+        self.patch_length = patch_length
+        self.stride = stride
+
+        # get the number of patches
+        self.num_patches = compute_num_patches(sequence_length, patch_length, stride)
+        new_sequence_length = patch_length + stride * (self.num_patches - 1)
+        self.s_begin = sequence_length - new_sequence_length
+
+    def forward(self, past_values: torch.Tensor):
+        """
+        Parameters:
```

Comment:
Docstring doesn't follow the right format

Reply:
It has been edited.